demo/dismissable/dismissable-item.js

```diff
@@ -0,0 +1,218 @@
+        break;
+      case 'touchmove':
+        this._onPointerPan(event, event.changedTouches[0]);
+        break;
+      case 'touchend':
+        this._onPointerUp(event, event.changedTouches[0]);
+        break;
+    }
+  }
+
+  setPosition(position) {
+    this.width = this.offsetWidth;
+    this.style.opacity = (this.width - Math.abs(position)) / this.width;
+    const currentY = this.style.transform.split(',')[1];
+    this.style.transform = `translate(${position}px, ${currentY}`;
+  }
+
+  _dismiss() {
+    this.style.opacity = 0;
+
+    const collapseAnim = this.animate({
+      height: [ getComputedStyle(this).height, '0px']
+    }, {
+      duration: 100,
+      iterations: 1
+    });
+
+    collapseAnim.onfinish = this._fireRemove.bind(this);
+  }
+
+  _fireRemove() {
+    this.setPosition(0);
+
+    const event = new CustomEvent('remove', {
+      detail: { itemIndex: this.itemIndex },
+      bubbles: true
+    });
+    this.dispatchEvent(event);
+  }
+
+  settle(targetPosition) {
+    this.state = 'initial';
+    if (targetPosition === this.position) {
+      return;
+    }
+
+    const currentY = this.style.transform.split(',')[1];
+    const isDismiss = targetPosition !== 0;
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, isDismiss ? 0 : 1 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) * 0.5,
+      iterations: 1
+    });
+
+    this.position = targetPosition;
+    animation.onfinish = () => isDismiss ? this._dismiss() : this.setPosition(0);
+  }
+
+  fling(velocityX) {
+    this.state = 'initial';
+    const targetPosition = velocityX < 0 ? -this.width : this.width;
+
+    const currentY = this.style.transform.split(',')[1];
+
+    const animation = this.animate({
+      transform: [
+        `translate(${this.position}px,${currentY}`,
+        `translate(${targetPosition}px,${currentY}`
+      ],
+      opacity: [ this.style.opacity, 0 ]
+    }, {
+      duration: Math.abs(targetPosition - this.position) / Math.abs(velocityX),
+      iterations: 1
+    });
+
+    animation.onfinish = this._dismiss.bind(this);
+  }
+
+  _settleToClosestPosition() {
+    const fraction = this.position / this.width;
+    if (fraction > 0.5) {
+      this.settle(this.width);
+    } else if (fraction < -0.5) {
+      this.settle(-this.width);
+    } else {
+      this.settle(0);
+    }
+  }
+
+  _onPointerDown(e, change) {
+    this.state = 'initial';
+    this.startX = change.clientX;
+    this.startY = change.clientY;
+    this.startPosition = 0;
+  }
+
+  _onPointerPan(e, change) {
+    if (this.state == 'initial') {
+      const deltaX = change.clientX - this.startX;
+      const deltaY = change.clientY - this.startY;
+
+      if (deltaX ** 2 + deltaY ** 2 < kTouchSlopValue ** 2) {
+        e.preventDefault();
```

Comment:
If I understood this correctly, we prevent the default if there was not enough movement. Why is this needed?

I would rather check if we are scrolling vertically or horizontally, and prevent default if really needed thereafter, something like this maybe:
```js
_onTouchStart(event) {
   // ...
   this.isHorizontalSwipe = null;
}
_onTouchMove(event) {
  if (this.isHorizontalSwipe === null) {
    const deltaX = Math.abs(change.clientX - this.startX);
    const deltaY = Math.abs(change.clientY - this.startY);
    // not enough movement to determine direction!
    if (deltaX < kTouchSlopValue  && deltaY < kTouchSlopValue) {
      return;
    }
    this.isHorizontalSwipe = deltaX >= deltaY;
  }
  // scrolling vertically, ignore this and following events
  if (this.isHorizontalSwipe === false) {
    return;
  }
  // scrolling horizontally!
```

Reply:
Sloppy or shaky fingers are quite common

Reply:
This is also what is done on Android and we did the same when I worked at Nokia :-)

You can see some examples here:
https://developer.android.com/training/gestures/viewgroup